scanner/src/test/kotlin/scanners/fossid/FossIdConfigTest.kt

```diff
@@ -113,6 +113,44 @@ class FossIdConfigTest : WordSpec({
             shouldThrow<IllegalArgumentException> { FossIdConfig.create(scannerConfig) }
         }
     }
+
+    "createNamingProvider" should {
+        "create a naming provider with a correct project naming convention" {
+            val scannerConfig = mapOf(
+                "serverUrl" to SERVER_URL,
+                "apiKey" to API_KEY,
+                "user" to USER,
+                "namingProjectPattern" to "#projectName_\$Org_\$Unit",
+                "namingVariableOrg" to "TestOrganization",
+                "namingVariableUnit" to "TestUnit"
+            ).toScannerConfig()
+
+            val fossIdConfig = FossIdConfig.create(scannerConfig)
+            val namingProvider = fossIdConfig.createNamingProvider()
+
+            val projectName = namingProvider.createProjectCode("TestProject")
```

Comment:
IMH you are mixing the tests here. A config test should only be about retrieving the configuration.
The call to `createProjectCode` and the verification of its return value should be in a `namingConventionTest.kt` or something.
Same comment underneath.

Reply:
The purpose of this test is to check whether a correct naming provider is created (which is functionality of `FossIdConfig`). The naming provider does not allow checking its internal properties, so the only way to check its initialization is to invoke it to generate names. It is true that these tests could be done in a separate test class, but testing this here was the easiest way to achive this.

Reply:
I think the solution here is to mark the properties of the `NamingProvider`  as `internal` so you can test them.
But the decision is up to you I won't fight about it.

Reply:
For `FossIdNamingProvider` this would be possible, but for `FossIdRestService`, I see no easy way how this could be done.